packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,358 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'localizations.dart';
+import 'picker.dart';
+
+/// Default aesthetic values obtained by comparing with iOS pickers.
+const double _kItemExtent = 32.0;
+const double _kPickerWidth = 330.0;
+/// Considers setting the default background color from the theme, in the future.
+const Color _kBackgroundColor = CupertinoColors.white;
+
+
+/// The iOS date picker has its width fixed to [_kPickerWidth] in all modes.
+/// If the maximum width given to the picker is greater than [_kPickerWidth],
+/// the leftmost and rightmost column will be extended equally so that the
+/// widths match, and the picker is in the center.
+///
+/// If the maximum width given to the picker is smaller than [_kPickerWidth],
+/// the picker's layout will be broken.
+
+
+/// A countdown timer picker in iOS style.
+///
+/// This picker shows duration as hour, minute and second spinners. The duration
```

Comment:
This picker shows a countdown duration with hour, minute and second spinners. The duration is bound between 0 and 23 hours 59 minutes 59 seconds.

Reply:
fixed. Much better!